### PR TITLE
Cipher type cleanup for aes

### PR DESCRIPTION
**Pull request by pabuhler**
This applies #170 & #172  on to master as well as making more consistent use of the srtp AES_XXX_ICM defines.

### CODE REVIEW

File: include/srtp.h [hunk 1]
```diff
@@ -1046,6 +1046,95 @@ void srtp_crypto_policy_set_aes_cm_256_hmac_sha1_32(srtp_crypto_policy_t *p);
  */
 void srtp_crypto_policy_set_aes_cm_256_null_auth(srtp_crypto_policy_t *p);
```
Comment (paulej) [inline include/srtp.h]: I'm all for keeping the API consistent, but I will note that here we have aes_cm_xxx and the variable names inside are aes_icm_xxx or AES_XXX_ICM.  Would be nice if those were all consistent.

File: crypto/cipher/aes_icm.c [hunk 2]
```diff
@@ -498,8 +501,21 @@ const srtp_cipher_type_t srtp_aes_icm = {
...
+const srtp_cipher_type_t srtp_aes_icm_256 = {
+    srtp_aes_icm_alloc,
+    srtp_aes_icm_dealloc,
+    srtp_aes_icm_context_init,
+    0,                          /* set_aad */
+    srtp_aes_icm_encrypt,
+    srtp_aes_icm_encrypt,
+    srtp_aes_icm_set_iv,
+    0,                          /* get_tag */
+    srtp_aes_icm_256_description,
```
Comment (JonathanLennox) [inline crypto/cipher/aes_icm.c]: Suggest renaming srtp_aes_icm_test_case_0 as srtp_aes_icm_128_test_case_0, and srtp_aes_icm_test_case_1 as srtp_aes_icm_256_test_case_0.

Comment (JonathanLennox) [inline crypto/cipher/aes_icm.c]: (This would match the names in aes_icm_ossl.c.)

File: srtp/srtp.c [hunk 3]
```diff
@@ -3081,6 +3098,53 @@ srtp_crypto_policy_set_aes_cm_256_null_auth (srtp_crypto_policy_t *p)
 }
 
 #ifdef OPENSSL
+void
+srtp_crypto_policy_set_aes_cm_192_hmac_sha1_80(srtp_crypto_policy_t *p) {
```
Comment (paulej) [inline srtp/srtp.c]: Any reason this function isn't sitting near the other functions with similar names?

File: crypto/cipher/aes_icm.c [hunk 4]
```diff
@@ -483,7 +483,7 @@ static const srtp_cipher_test_case_t srtp_aes_icm_test_case_1 = {
     0,
     NULL,
     0,
-    &srtp_aes_icm_test_case_0                 /* pointer to next testcase */
+    NULL,                 /* pointer to next testcase */
```
Comment (JonathanLennox) [inline crypto/cipher/aes_icm.c]: Separating the aes_128 and aes_256 test cases.

File: crypto/cipher/aes_icm.c [hunk 5]
```diff
@@ -402,7 +401,8 @@ static srtp_err_status_t srtp_aes_icm_encrypt (void *cv,
     return srtp_err_status_ok;
 }
 
-static const char srtp_aes_icm_description[] = "aes integer counter mode";
+static const char srtp_aes_icm_128_description[] = "AES-128 integer counter mode";
+static const char srtp_aes_icm_256_description[] = "AES-256 integer counter mode";
```
Comment (JonathanLennox) [inline crypto/cipher/aes_icm.c]: libsrtp's built-in aes implementation doesn't implement aes-192.

File: crypto/cipher/aes_icm_ossl.c [hunk 6]
```diff
@@ -460,7 +460,7 @@ static const srtp_cipher_test_case_t srtp_aes_icm_256_test_case_2 = {
  * This is the function table for this crypto engine.
  * note: the encrypt function is identical to the decrypt function
  */
-const srtp_cipher_type_t srtp_aes_icm = {
+const srtp_cipher_type_t srtp_aes_icm_128 = {
```
Comment (JonathanLennox) [inline crypto/cipher/aes_icm_ossl.c]: They were already defined for the aes_icm_ossl version.

File: crypto/cipher/aes_icm.c (hunk 5 above)
Comment (paulej) [inline crypto/cipher/aes_icm.c]: Oh, well that's a good reason. I had it in my head that we had only disabled 192-bit keys for BoringSSL, but we now have that via #209.  While I don't particularly see the value in 192-bit key lengths, myself, I sure do like consistency.

File: crypto/cipher/aes_icm.c (hunk 2 above)
Comment (pabuhler) [inline crypto/cipher/aes_icm.c]: :+1:

File: crypto/cipher/aes_icm.c (hunk 5 above)
Comment (pabuhler) [inline crypto/cipher/aes_icm.c]: @paulej well you could always implement it ...

File: crypto/cipher/aes_icm.c [hunk 7]
```diff
@@ -501,8 +501,21 @@ const srtp_cipher_type_t srtp_aes_icm = {
     srtp_aes_icm_encrypt,
     srtp_aes_icm_set_iv,
     0,                          /* get_tag */
-    srtp_aes_icm_description,
-    &srtp_aes_icm_test_case_1,
-    SRTP_AES_ICM
+    srtp_aes_icm_128_description,
+    &srtp_aes_icm_test_case_0,
+    SRTP_AES_128_ICM
```
Comment (pabuhler) [inline crypto/cipher/aes_icm.c]: I was in general just following the existing naming conventions, will look at consolidating one way or the other.

File: srtp/srtp.c (hunk 3 above)
Comment (pabuhler) [inline srtp/srtp.c]: I would guess it was placed there due to the opeenssl block

File: crypto/cipher/aes_icm.c (hunk 5 above)
Comment (paulej) [inline crypto/cipher/aes_icm.c]: If I had time, it would be fun to do.

File: include/srtp.h [hunk 8]
```diff
@@ -1046,6 +1046,95 @@ void srtp_crypto_policy_set_aes_cm_256_hmac_sha1_32(srtp_crypto_policy_t *p);
...
+ * @brief srtp_crypto_policy_set_aes_cm_192_hmac_sha1_32() sets a crypto
+ * policy structure to a short-authentication tag policy using AES-192
+ * encryption.
+ *
+ * @param p is a pointer to the policy structure to be set
+ *
+ * The function call srtp_crypto_policy_set_aes_cm_192_hmac_sha1_32(&p)
+ * sets the crypto_policy_t at location p to use policy
+ * AES_CM_192_HMAC_SHA1_32 as defined in
+ * draft-ietf-avt-srtp-big-aes-03.txt.  This policy uses AES-192
```
Comment (paulej) [inline include/srtp.h]: This is now RFC 6188.  Not sure how many such references like this exist in the code.

Comment (pabuhler) [inline include/srtp.h]: we should try to do a review of references in general, I think there are a few that are incorrect

**Pull request merged**